### PR TITLE
Add Code Review guidelines

### PR DESCRIPTION
**Pull request by marcofucci**
### Description of change

This formalises the first part of the code review process as recently discussed and agreed. 

When approved, an identical version will be created in the [frontend](https://github.com/uktrade/data-hub-frontend) repo.

The remaining steps (*Step 2: PR submitted* and *Step 3: PR approved*) will be added after being discussed.

### CODE REVIEW

File: docs/Code review guidelines.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+- Document what the PR does and why the change is needed
+- Give full context - not everyone has access to Jira/Trello
+- Detail anything that is out of scope and will be covered by future PRs
+- Include details on the lifecycle of the feature and its nature. Is it a new feature or a change to an existing one? Is the code going to be short-lived? Is it part of a bigger piece of work?
+- Highlight possible controversies
+- Include instructions on how to test (e.g. what should I see?)
+- Detail any considerations when releasing
+
+### Screenshots
+- Add before / after screenshots or GIFs
```
Comment (jim68000) [inline docs/Code review guidelines.md]: These bits relevant for a BE PR guideline, or do we want to share this across teams?

File: docs/Code review guidelines.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Code Review guidelines
+
+*Discussed and agreed by all Data Hub developers. If anything changes, please keep this document up-to-date and in sync with the related backend/frontend one.*
+
+## Step 1: Before and when submitting a PR
+
+### Code and tests
+
+- Make sure your code is appropriately documented and includes change log fragments describing the change
+- Keep the overall coding style consistent with the rest of the repository (within limits). When adopting a new agreed style (ideally) refactor the existing code to keep the overall codebase consistent
```
Comment (jim68000) [inline docs/Code review guidelines.md]: Might be worth expanding on what '(within limits)' means as i think there have been a lot of PRs which have got stuck on this point

File: docs/Code review guidelines.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+### Commit Hygiene
+- Follow branch naming conventions by prefixing your branch name with *feature/*, *bugfix/*, *removal/*, *test/*, *hotfix/* or *release/*
+- Make sure commits are logical and atomic - each commit should include tests
+- Keep each commit small and deployable - each commit should ideally leave the develop branch in a releasable state
+- Use [imperative mood](https://git.kernel.org/pub/scm/git/git.git/tree/Documentation/SubmittingPatches?id=HEAD#n133) in commit message but it’s okay to use past tense in the description
+
+### PR Hygiene
+- Make sure your PR is atomic and doesn’t solve multiple problems at the same time
+- Keep your PR small and deployable - each PR **must** leave the develop branch in a releasable state
+- The ideal size for a PR is [400 lines of code](https://smartbear.com/learn/code-review/best-practices-for-peer-code-review/)
```
Comment (jim68000) [inline docs/Code review guidelines.md]: Is than ideal or the ideal _maximum_ length? I would have thought a one liner was ideal ;-)

File: docs/Code review guidelines.md (hunk 1 above)
Comment (marcofucci) [reply docs/Code review guidelines.md]: Not always but they do make sense when changing the django admin for example so worth keeping them.

**Pull request merged**